File: projects/order-creation/core/src/main/java/com/robinfood/core/dtos/transactionrequestdto/CouponDTO.java
package com.robinfood.core.dtos.transactionrequestdto;

import java.io.Serializable;
import java.math.BigDecimal;
import javax.validation.constraints.NotBlank;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class CouponDTO implements Serializable {

    private static final long serialVersionUID = 60512316832949847L;

    @NotBlank
    private String code;

    private BigDecimal value;

}

File: projects/posv2/core/src/main/kotlin/com/robinfood/core/dtos/transactionrequest/CouponDTO.kt
package com.robinfood.core.dtos.transactionrequest

import java.math.BigDecimal
import javax.validation.constraints.Min
import javax.validation.constraints.NotBlank
import javax.validation.constraints.NotNull

data class CouponDTO(

        @NotBlank(message = "Coupon code must not be empty")
        val code: String,

        @Min(0)
        @NotNull
        val value: BigDecimal
)